providers/default.rb

```diff
@@ -1,16 +1,30 @@
 use_inline_resources
 
 action :update do
+  dir        = new_resource.base_name.split('/')[0..-2].join('/')
+  push_event = new_resource.push_to_datadog || false
+  event_name = new_resource.datadog_event_name
+  event_text = new_resource.datadog_event_text
+  event_key  = new_resource.datadog_event_key
+
```

Comment:
We seem to do it this way in a number of cookbooks, so I assumed that it's the way to go. And I showed @slantin some code where they did it this way.